cross-modal-search/app.py

```diff
@@ -26,53 +26,56 @@ def input_index_data(num_docs=None, batch_size=8, dataset='f30k'):
                                   split='test',
                                   batch_size=batch_size,
                                   dataset_type=dataset)
+
     for i, (images, captions) in enumerate(data_loader):
+        print(f' images {len(images)}, captions {len(captions)}')
         for image in images:
-            document = jina_pb2.Document()
-            document.buffer = image
-            document.modality = 'image'
-            document.mime_type = 'image/jpeg'
+            with Document() as document:
+                document.buffer = image
+                document.modality = 'image'
+                document.mime_type = 'image/jpeg'
             yield document
 
         for caption in captions:
-            document = jina_pb2.Document()
-            document.text = caption
-            document.modality = 'text'
-            document.mime_type = 'plain/text'
+            with Document() as document:
+                document.text = caption
+                document.modality = 'text'
+                document.mime_type = 'text/plain'
             yield document
 
         if num_docs and (i + 1) * batch_size >= num_docs:
             break
 
 
 def input_search_text_data(text):
-    document = jina_pb2.Document()
+    document = Document()
     document.text = text
     document.modality = 'text'
-    document.mime_type = 'plain/text'
+    document.mime_type = 'text/plain'
     return [document]
 
 
 def input_search_image_file(image_file_path):
     with open(image_file_path, 'rb') as fp:
         image_buffer = fp.read()
-    document = jina_pb2.Document()
+    document = Document()
```

Comment:
I think you need to do `with Document` in order to get the `content_hash`. Or call `update_content_hash` yourself. It doesn't matter for this example, but would be good that people see it when they use the example.

Reply:
Okey, will use ctxt management